### DISCUSSION TITLE
show required options on usage

### DISCUSSION THREAD

**Discussion by scarf005** [c4spar/cliffy#583] [Ideas]
since they are always required, it'd be great for usage to also display required options.

example:
```ts
  Usage: COMMAND <id> --replace <type> --using <type>

  Description:

    replace contents of a JSON entry with another JSON entry with the same id.   
    this is useful for replacing mod content with vanilla content and vice versa.

  Options:

    -h, --help             - Show this help.                                 
    -l, --lint             - lint all json files after migration.            
    -r, --replace  <type>  - path to recursively search jsons.     (required)
    -u, --using    <type>  - path to recursively search jsons.     (required)
```

**Comment by c4spar** [comment]
This is already possible. You can use the `hints` option from the `.help()` method for this:

```ts
new Command()
  .help({
    hints: true,
  })
```

**Reply by scarf005** [reply]
...oh, i meant generating `Usage` section automatically. for example, as `replace` and `using` are required they could be automatically included in `Usage`.

`COMMAND <id> --replace <type> --using <type>`

**Reply by c4spar** [reply]
ah sry now i understood. yeah that could make sense. good idea 👍
